dbt/models/default/default.vw_pin_value.sql

```diff
@@ -1,190 +1,472 @@
+                    THEN asmt.valapr3
             END
         ) AS board_tot_mv
-    FROM {{ source('iasworld', 'asmt_all') }}
-    WHERE procname IN ('CCAOVALUE', 'CCAOFINAL', 'BORVALUE')
-        AND rolltype != 'RR'
-        AND deactivat IS NULL
-        AND valclass IS NULL
-    GROUP BY parid, taxyr
+    FROM {{ source('iasworld', 'asmt_all') }} AS asmt
+    LEFT JOIN stages
+        ON asmt.parid = stages.parid
+        AND asmt.taxyr = stages.taxyr
+    WHERE (
+        -- Check for two possible situations: Either the record is stamped
+        -- with a procname corresponding to a stage, in which case the record
+        -- corresponds to a final value for a stage, or the procname is null
+        -- and other fields indicate that it is a provisional value for an
+        -- upcoming stage
+        asmt.procname IN ('CCAOVALUE', 'CCAOFINAL', 'BORVALUE')
+        OR (
+            asmt.procname IS NULL
+            AND asmt.cur = 'Y'
+            AND (
+                -- If the PIN has no stages but its year is not the current
+                -- assessment year, it is likely a data error from a prior
+                -- year that we don't want to include in our results. In
+                -- contrast, if the PIN is in the current year but has no
+                -- stages, it is most likely a provisional value for a PIN
+                -- that has not mailed yet
+                CARDINALITY(stages.procnames) != 0
+                OR asmt.taxyr = DATE_FORMAT(NOW(), '%Y')
+            )
+        )
+    )
+    AND asmt.rolltype != 'RR'
+    AND asmt.deactivat IS NULL
+    AND asmt.valclass IS NULL
+    GROUP BY asmt.parid, asmt.taxyr
 ),
 
 clean_values AS (
     SELECT
         stage_values.*,
         -- Current stage indicator
         CASE
-            WHEN stage_values.board_tot IS NOT NULL THEN 'BOARD CERTIFIED'
+            WHEN
+                stage_values.board_tot IS NOT NULL
+                THEN '{{ stage_name_board }}'
             WHEN
                 stage_values.certified_tot IS NOT NULL
-                THEN 'ASSESSOR CERTIFIED'
-            WHEN stage_values.mailed_tot IS NOT NULL THEN 'MAILED'
+                THEN '{{ stage_name_certified }}'
+            WHEN
+                stage_values.pre_certified_tot IS NOT NULL
+                THEN '{{ stage_name_pre_certified }}'
+            WHEN
+                stage_values.mailed_tot IS NOT NULL
+                THEN '{{ stage_name_mailed }}'
+            WHEN
+                stage_values.pre_mailed_tot IS NOT NULL
+                THEN '{{ stage_name_pre_mailed }}'
         END AS stage_name,
         CASE
             WHEN stage_values.board_tot IS NOT NULL THEN 3
             WHEN stage_values.certified_tot IS NOT NULL THEN 2
+            WHEN stage_values.pre_certified_tot IS NOT NULL THEN 1.5
             WHEN stage_values.mailed_tot IS NOT NULL THEN 1
+            WHEN stage_values.pre_mailed_tot IS NOT NULL THEN 0
```

Comment:
I think `0` makes a lot of sense for the pre-mail stage number, but `1.5` feels kind of weird for pre-certified. Is it risky to increment these so that they run `1-5` (or `0-4`)?